src/wcl/tracing.cpp

```diff
@@ -138,11 +138,41 @@ void FormatSubscriber::receive(const Event& e) {
     s << item.first << "=" << item.second;
   }
 
-  s << "]";
+  s << "] ";
 
   if (auto* value = e.get(LOG_MESSAGE)) {
-    s << " " << *value;
+    s << *value;
+  } else {
+    s << "<empty message>";
   }
+
+  s << std::endl;
+}
+
+void UrgentSubscriber::receive(const Event& e) {
```

Comment:
I *really really really* think we should just have a PredicateSubscriber like I've been asking for it covers so many use cases and its so simple to implement.

```
wcl::log::subscribe(std::make_unique<wcl::log::FormatSubscriber>(log_file.rdbuf()));
auto cerr_sub = std::make_unique<wcl::log::FormatSubscriber>(std::cerr.rdbuf());
auto urgent_sub =  std::make_unique<wcl::log::FilterSubscriber>(std::move(cerr_sub)[](const auto& event) {
  return event[LOG_URGENT] == "1";
});
wcl::log::subscribe(std::move(urgent_sub));
```

This gives us nearly everything we could ever want to do without writing one off specific subscribers and the subscriber behavior is apparent at the subscriber site. I know last time you said you wanted a simpler output format but I think that's orthogonal:

```
wcl::log::subscribe(std::make_unique<wcl::log::FormatSubscriber>(log_file.rdbuf()));
auto cerr_sub = std::make_unique<wcl::log::SimpleFormatSubscriber>(std::cerr.rdbuf());
auto urgent_sub =  std::make_unique<wcl::log::FilterSubscriber>(std::move(cerr_sub)[](const auto& event) {
  return event[LOG_URGENT] == "1";
});
wcl::log::subscribe(std::move(urgent_sub));
```

As discussed prior though I would be ok with moving this into the tools instead.